tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
+        if not response.ok:
+            raise tmt.utils.ReportError(
+                f"\nReceived non-ok status code from ReportPortal: {response.text}")
+
+        self.debug("Response code from the server", str(response.status_code))
+        self.debug("Message from the server", str(response.text))
+
+    TMT_TO_RP_RESULT_STATUS = {
+        ResultOutcome.PASS: "PASSED",
+        ResultOutcome.FAIL: "FAILED",
+        ResultOutcome.ERROR: "FAILED",
+        ResultOutcome.WARN: "FAILED",
+        ResultOutcome.INFO: "SKIPPED"
+        }
+
     def go(self) -> None:
         """
-        Read executed tests, prepare junit, compress it to a zip zile and
-        send it to the ReportPortal instance.
+        Create a ReportPortal launch and its test items,
+        fill it with all parts needed and report the logs.
         """
 
         super().go()
 
-        # Check the data, show interesting info to the user
-        # Required fields are: url, token and project
         server = self.get("url")
         if not server:
             raise tmt.utils.ReportError("No ReportPortal server url provided.")
         server = server.rstrip("/")
 
+        project = self.get("project")
+        if not project:
+            raise tmt.utils.ReportError("No ReportPortal project provided.")
+
         token = self.get("token")
         if not token:
             raise tmt.utils.ReportError("No ReportPortal token provided.")
 
-        project = self.get("project")
-        if not project:
-            raise tmt.utils.ReportError("No ReportPortal project provided.")
-        self.info("project", project, color="green")
-
-        # Use provided launch name, default to run workdir name
-        assert self.step.plan.my_run is not None
-        assert self.step.plan.my_run.workdir is not None
-        launch_name = self.get("launch-name") or self.step.plan.my_run.workdir.name
-        self.info("launch", launch_name, color="green")
-
-        # Generate a xUnit report
-        import_junit_xml()
-        assert junit_xml is not None
-        suite = junit.make_junit_xml(self)
-        data = junit_xml.TestSuite.to_xml_string([suite])
-
-        # Zip the report
-        bytestream = io.BytesIO()
-        # SIM117: Use a single `with` statement with multiple contexts instead of nested `with`
-        # statements. Here, `zipstream` is being used in the second nested context.
-        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,  # noqa: SIM117
-                             compresslevel=1) as zipstream:
-            # XML file names are irrelevant to ReportPortal
-            with zipstream.open("tests.xml", "w") as entry:
-                entry.write(data.encode("utf-8"))
-        bytestream.seek(0)
-
-        # Send the report to the ReportPortal instance
+        assert self.step.plan.name is not None
+        launch_name = self.get("launch") or self.step.plan.name
+
+        version = "v1"
```

Comment:
This smells like a magic constant, ti would deserve to be a module-level name, e.g. `DEFAULT_API_VERSION` or something similar, to be easier to hunt down when things change and this becomes `v2`.

Reply:
Good point, I typed it temporary and forgot about that. Although was not sure about the suitable name for it. As there is `v1` for most of use-cases and then `v2` for asynchronous reporting. Let's call it `RP_API_VERSION` for now.